modules/cart/src/CartSession.php

```diff
@@ -63,6 +71,10 @@ public function deleteCartId($cart_id) {
       // Remove the empty list to allow the system to clean up empty sessions.
       $this->session->remove('commerce_cart_orders');
     }
+    // Keep a list of completed order for access checks to completed orders.
+    $completed_ids = $this->session->get('commerce_cart_completed_orders', []);
+    $completed_ids = array_merge($completed_ids, [$cart_id]);
+    $this->session->set('commerce_cart_completed_orders', $completed_ids);
```

Comment:
So should this just go in `::finalizeCart` and make sure `::finalizeCart` doesn't invoke delete? Seems weird we have `deleteCartId` but its actually doing a conversions.